### PR TITLE
Added a table for problem recommendations.

### PR DESCRIPTION
**Pull request by MipsaPatel**
Added a table for problem recommendations.

### CODE REVIEW

File: models/db.py [hunk 1]
```diff
@@ -776,6 +776,13 @@ def _count_custom_users_lambda(row):
                 Field("contest_id"),
                 Field("name"))
 
+db.define_table("problem_recommendations",
+                Field("user_id", "reference auth_user"),
+                Field("problem_id", "reference problem"),
+                Field("state", "integer"),
```
Comment (raj454raj) [inline models/db.py:782]: Can you please document the states in the PR

Comment (MipsaPatel) [reply models/db.py:782]: To confirm, add a comment in the code and not this PR, right?

Comment (raj454raj) [reply models/db.py:782]: Yep, in the code

File: models/db.py [hunk 2]
```diff
@@ -776,6 +776,15 @@ def _count_custom_users_lambda(row):
                 Field("contest_id"),
                 Field("name"))
 
+db.define_table("problem_recommendations",
+                Field("user_id", "reference auth_user"),
+                Field("problem_id", "reference problem"),
+                # The possible states of a recommendation are: 0 - Recommended,
+                # 1 - Viewed, 2 - Attempted, 3 - Solved
+                Field("state", "integer",
```
Comment (raj454raj) [inline models/db.py:784]:
```suggestion
                # The possible states of a recommendation are:
                # 0 - Recommended, 1 - Viewed, 2 - Attempted, 3 - Solved
                Field("state", "integer", default=0),
```

**Pull request merged**